docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+Finished in 0.034 seconds
+1 example, 0 failures
+----------------------------------
+
+Hooray! You're almost there! (Unless you saw failures... you should fix those
+  first).
+
+=== building the gem and testing via logstash
+Now you're ready to build your (well-tested) plugin into a Ruby gem. You already
+have all the necessary ingredients, if you've been following along, so let's go
+ahead and run the build command:
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+gem build logstash-{plugintype}-{pluginname}.gemspec
+----------------------------------
+
+Download the latest version from the
+http://www.elasticsearch.org/overview/logstash/download/[Logstash downloads page].
+Untar and cd in to the directory:
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+curl -O http://download.elasticsearch.org/logstash/logstash/logstash-{ls_version}.tar.gz
+tar xzvf logstash-{ls_version}.tar.gz
+cd logstash-{ls_version}
+----------------------------------
+
+Let's explore the Logstash plugin tool to determine which plugins are currently
+available:
+
+[source,sh]
+----------------------------------
+bin/plugin list
+----------------------------------
+Depending on the version of Logstash you have installed, you might see a short
+or long list of plugins: inputs, codecs, filters and outputs.
+
+Using this plugin too, we can install the gem we just built (use the correct
+  path to the gem for your environment):
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+bin/plugin install /my/logstash/plugins/logstash-{plugintype}-{pluginname}/logstash-{plugintype}-{pluginname}.gem
+----------------------------------
+
+After running this, you should see feedback from Logstash that it was
+successfully installed:
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+validating /my/logstash/plugins/logstash-{plugintype}-{pluginname}/logstash-{plugintype}-{pluginname}-0.1.1.gem >= 0
+Valid logstash plugin. Continuing...
+Successfully installed 'logstash-{plugintype}-{pluginname}' with version '0.1.1'
+----------------------------------
+
+Now try running Logstash with a simple configuration passed in via the
+command-line, using the `-e` flag.
+
+ifdef::run_method[]
+[source,sh]
+[subs="attributes"]
+----------------------------------
+bin/logstash -e 'input { {pluginname}{} } output {stdout { codec => rubydebug }}'
+----------------------------------
+
+Type something, in this case "it worked!", and you should see the new message,
+"it worked!" output by Logstash:
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+Using milestone 1 {plugintype} plugin '{pluginname}'. This plugin should work, but would benefit from use by folks like you. Please let us know if you find bugs or have suggestions on how to improve this plugin.  For more information on plugin milestones, see http://logstash.net/docs/{ls_version}/plugin-milestones {:level=>:warn}
```

Comment:
we will have a differently worded warning after @ph merged https://github.com/elasticsearch/logstash/pull/2331

See https://github.com/elasticsearch/logstash/blob/master/locales/en.yml#L50